### ISSUE TITLE
A TV Shows folder inside the TV Shows folder

### ISSUE THREAD

**Issue by rqchin** [mackworth/cTiVo#388]
Somewhere I messed up in selecting/reselecting the root folder for cTivo to use and now I've ended up with a TV Shows folder (with a hierarchy of folders with shows in them) inside my existing TV Shows directory.

Is it safe for me to:
- change the default back to the parent TV Shows folder (assuming I get the interface right and select the correct parent folder)
- Quit cTivo
- Move files of recorded shows to their similar folder in the top level TV Shows folder
- Launch cTvio again

or would this mess up cTivo somehow? Or is there an automated way to merge this more gracefully that keeps cTivo updated?

-Rick

**Comment by mackworth** [comment]
Yes, that's all fine. cTiVo doesn't really "monitor" the folder structure or care how it's structured. What happens is that when starting a download, cTiVo simply checks the current template, and then makes sure that whatever directory (or path of directories) is needed to store the new video  is available, or else it creates it.  So whatever changes you make later won't affect any future behavior. You don't even have to quit, although such directory restructuring is obviously not recommended DURING a download!

Note from Apple:
> Merge two folders with the same name:
> If you have two folders with identical names at two different locations, you can merge them into a single folder. On your Mac, press and hold the Option key, then drag one folder to the location that contains a folder with the same name. In the dialog that appears, click Merge.

So if you have ~/Movies/TVShows/TVShows, you have three generations of folders; call them Subordinate(TVShows), Parent(TVShows). and Grandparent(Movies).
That would suggest you can drag the Subordinate to another folder (not under Grandparent, say Movies2), then option-drag the Subordinate back to the Grandparent. As long as Subordinate and Parent have the same name, Finder should offer the merge option, which will recursively merge all the folders together into Parent. It will leave the old Subordinate unchanged, so you can then delete it. (If you're on APFS, then there's no physical file copying; don't know if there is on HFS+)

**Comment by mackworth** [comment]
So, did that work for you?

**Comment by rqchin** [comment]
I haven't tried it yet but I will get a chance to in two weeks.

**Comment by rqchin** [comment]
OK, I was able to login remotely and run a test. I took a subset of the shows/folders and tried it and the merge process seemed to work as expected. My one concern is it resolves direct conflicts (two files in the same relative position in the hierarchy with the same name) by overwriting the older file strictly by date/time, but that's kind of understandable. I did a similar test with two folders I created with a mix of files and that merged as expected as well.
